File: Week 5 - 10. SimpleTabAppCustom/SimpleTabAppKai/SecondViewController.m
//
//  SecondViewController.m
//  SimpleTabAppKai
//
//  Created by  Julio Reyes  on 4/1/15.
//  Copyright (c) 2015 JulioReyesCorps. All rights reserved.
//

#import "SecondViewController.h"

@interface SecondViewController ()
@property (nonatomic, weak) IBOutlet UITextField *txtNotify;
@end

@implementation SecondViewController
- (instancetype)init
{
    self = [super init];
    if (self) {
        self.tabBarItem.image = [UIImage imageNamed:@"second"];
    }
    return self;
}
- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.title = @"Second";
}
- (void)viewWillAppear:(BOOL)animated {
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(keyboardWillShow:) name:UIKeyboardWillShowNotification object:nil];
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(keyboardWillHide:) name:UIKeyboardWillHideNotification object:nil];
    
    [[NSNotificationCenter defaultCenter] addObserver:self
                                             selector:@selector(receiveNotification:) name:@"Test1" object:nil];
    [[NSNotificationCenter defaultCenter] addObserver:self
                                             selector:@selector(receiveNotification:) name:@"Test2" object:nil];
}

- (void)viewWillDisappear:(BOOL)animated {
    [[NSNotificationCenter defaultCenter] removeObserver:self name:UIKeyboardWillShowNotification object:nil];
    [[NSNotificationCenter defaultCenter] removeObserver:self name:UIKeyboardWillHideNotification object:nil];
    [[NSNotificationCenter defaultCenter]removeObserver:self];
}
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

-(IBAction)onePressed:(id)sender
{
    NSDictionary *extraInfo = [NSDictionary dictionaryWithObject:@"One" forKey:@"button_name"];
    [[NSNotificationCenter defaultCenter] postNotificationName:@"Test1" object:self userInfo:extraInfo ];
}

-(IBAction)twoPressed:(id)sender
{
    NSDictionary *extraInfo = [NSDictionary dictionaryWithObject:@"Two" forKey:@"button_name"];
    [[NSNotificationCenter defaultCenter] postNotificationName:@"Test2" object:self userInfo:extraInfo ];
    
}


-(void)receiveNotification:(NSNotification *) notification
{
    NSLog(@"Second View Notification Received: %@", [notification name]);
    
    NSDictionary *extraInfo = [notification userInfo];
    [_txtNotify setText:[NSString stringWithFormat:@"%@ was pressed.", [extraInfo objectForKey:@"button_name"]]];
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/
- (void)keyboardWillShow:(NSNotification *)notification {
    CGSize keyboardSize = [[[notification userInfo] objectForKey:UIKeyboardFrameBeginUserInfoKey] CGRectValue].size;
    
    float newVerticalPosition = -keyboardSize.height;
    
    [self moveFrameToVerticalPosition:newVerticalPosition forDuration:0.3f];
}


- (void)keyboardWillHide:(NSNotification *)notification {
    [self moveFrameToVerticalPosition:0.0f forDuration:0.3f];
}


- (void)moveFrameToVerticalPosition:(float)position forDuration:(float)duration {
    CGRect frame = self.view.frame;
    frame.origin.y = position;
    
    [UIView animateWithDuration:duration animations:^{
        self.view.frame = frame;
    }];
}

- (BOOL)textFieldShouldReturn:(UITextField *)textField
{
    [textField resignFirstResponder];
    return YES;
}
- (void)touchesBegan:(NSSet *)touches withEvent:(UIEvent *)event{
    NSLog(@"touchesBegan:withEvent:");
    [self.view endEditing:YES];
    [super touchesBegan:touches withEvent:event];
}

@end

File: Swift/class 15/SwiftTabAppSample/SwiftTabAppSample/SecondViewController.swift
//
//  SecondViewController.swift
//  SwiftTabAppSample
//
//  Created by Oren Goldberg on 12/10/14.
//  Copyright (c) 2014 Turn To Tech. All rights reserved.
//

import UIKit

class SecondViewController: UIViewController, UITextFieldDelegate
 {

    @IBOutlet var txtNotify: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        NSNotificationCenter.defaultCenter().addObserver(self, selector: "receiveNotification:", name: "Test1", object: nil)
        
        NSNotificationCenter.defaultCenter().addObserver(self, selector: "receiveNotification:", name: "Test2", object: nil)
    }

    
    @IBAction func onePressed(sender: AnyObject) {
        var extraInfo:Dictionary = ["button_name":"One"]
        NSNotificationCenter.defaultCenter().postNotificationName("Test1", object: nil, userInfo: extraInfo)
        
    }
    
    @IBAction func twoPressed(sender: AnyObject) {
        var extraInfo:Dictionary = ["button_name":"Two"]
        NSNotificationCenter.defaultCenter().postNotificationName("Test2", object: nil, userInfo: extraInfo)

    }
    
    func receiveNotification(notification:NSNotification) {
        println("Second View Notification received: \(notification.name)")
        var extra_info:NSDictionary
        extra_info = notification.userInfo!
        txtNotify.text = extra_info["button_name"] as? String
    }
    
    
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

}